### PR TITLE
New addon: More costume fonts

### PR DESCRIPTION
**Pull request by valmontechno**
An add-on that adds the following fonts to the costume editor.
Scratch, Arial, Helvetica, Verdana, Impact and Monospace.

### CODE REVIEW

**Comment by BroJac5246** [comment]
Related:
#2050 (so this is _probably_ rejected)
#3059

File: addons/more-costume-fonts/userscript.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		{ 'name': 'Monospace', 'family': 'Monospace' }
+	]
+	
+	function setFont(font, update) {
+		paper.tool.changeFont(font.font);
+		document.querySelector("[class*='font-dropdown_mod-unselect']").click();
+		updateImage();
+	}
+	
+	function addNewFont(font, previousFont, modContextMenu) {
```
Comment (TheColaber) [inline addons/more-costume-fonts/userscript.js:29]: Can you just add a check for `if (!addon.self.disabled) return` at the top of this function and then add `dynamicEnable` and `dynamicDisable` in the addon.json, so the addon can be turned on and off dynamically?

File: addons/more-costume-fonts/userscript.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		const button = document.createElement('span');
+		button.className = addon.tab.scratchClass('button_button') + ' ' + addon.tab.scratchClass('font-dropdown_mod-menu-item');
```
Comment (TheColaber) [inline addons/more-costume-fonts/userscript.js:31]: I think you can just use `addon.tab.scratchClass('button_button', 'font-dropdown_mod-menu-item')`

File: addons/more-costume-fonts/userscript.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+		button.addEventListener('mouseover', function(){ paper.tool.changeFont(font.family); }, false);
+		button.addEventListener('mouseout', function(){ paper.tool.changeFont(previousFont); }, false);
+	}
+
+	while (true) {
+		const elem = await addon.tab.waitForElement("[class*='font-dropdown_mod-context-menu'], [class*='font-dropdown_displayed-font-name']", {
+			markAsSeen: true,
+		});
+		
+		if (elem.classList.contains(addon.tab.scratchClass('font-dropdown_mod-context-menu')) && elem.classList.contains(addon.tab.scratchClass('input-group_input-group'))) {
```
Comment (TheColaber) [inline addons/more-costume-fonts/userscript.js:47]: Won't the first part of this if statement always be true? `font-dropdown_mod-context-menu` is already in the waitForElement, and waitForElement should guarantee that class will exist on the element.

File: addons/more-costume-fonts/addon.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
+{
+	"name": "More costume fonts",
+	"description": "Added new fonts for the costume editor.",
+	"tags": ["editor", "costumeEditor"],
+	"info": [
+		{
+			"type": "notice",
+			"text": "Some fonts may not be compatible with all devices.",
```
Comment (TheColaber) [inline addons/more-costume-fonts/addon.json:8]: Could you tell us what they will and won't work with and why?

File: addons/more-costume-fonts/userscript.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+			elem.textContent = '';
```
Comment (TheColaber) [inline addons/more-costume-fonts/userscript.js:48]: Is it really necessary to clear the element's previous items and then re-add them in?

File: addons/more-costume-fonts/userscript.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		{ 'name': 'Pixel', 'family': 'Pixel' },
+		{ 'name': '中文', 'family': '"Microsoft YaHei", "微软雅黑", STXihei, "华文细黑"' },
+		{ 'name': '日本語', 'family': '"ヒラギノ角ゴ Pro W3", "Hiragino Kaku Gothic Pro", Osaka, "メイリオ", Meiryo, "ＭＳ Ｐゴシック", "MS PGothic"' },
+		{ 'name': '한국어', 'family': 'Malgun Gothic' },
+		{ 'name': 'Scratch', 'family': 'Scratch' },
+		{ 'name': 'Arial', 'family': 'Arial' },
+		{ 'name': 'Helvetica', 'family': 'Helvetica' },
+		{ 'name': 'Verdana', 'family': 'Verdana' },
+		{ 'name': 'Impact', 'family': 'Impact' },
+		{ 'name': 'Monospace', 'family': 'Monospace' }
```
Comment (TheColaber) [inline addons/more-costume-fonts/userscript.js:20]: Are there any other fonts that work natively in the scratch editor? How can we find out?

**Review by TheColaber — CHANGES_REQUESTED** [review]

**Comment by Secret-chest** [comment]
@valmontechno @BroJac5246 @TheColaber Remember not all devices are guaranteed to have these fonts (example Linux)!

**Comment by Samq64** [comment]
> Remember not all devices are guaranteed to have these fonts (example Linux)!

Aa a linux user I think it's fine as long as there's a notice like we alreay do for the more keys addon.

**Comment by Secret-chest** [comment]
> > Remember not all devices are guaranteed to have these fonts (example Linux)!
> 
> As a linux user I think it's fine as long as there's a notice like we alreay have for the more keys addon.

I don't want to have to see text that is cut off!

**Comment by Secret-chest** [comment]
Also Scratch will get removed.

**Comment by mxmou** [comment]
This doesn't solve any of the problems previous suggestions for custom fonts had. These fonts (except Scratch) only work if they're installed on the user's device - most of them are only available on Windows and macOS. As @Secret-chest pointed out, a fallback font will be used on other platforms, which sometimes causes text to be cut off or otherwise broken.

Monospace is a generic font name that can refer to any monospace font, which causes similar issues.

**Comment by WorldLanguages** [comment]
This is probably rejected, so I'm closing this PR to avoid any misunderstandings.

**Pull request closed without merging**